Hippo.Core/Services/AccountSyncService.cs

```diff
@@ -119,19 +197,15 @@ private async Task AddNewGroupAccounts(Cluster cluster)
                 .Where(pgpu => pgpu.ClusterName == cluster.Name && _dbContext.Groups.Any(group => group.IsActive && group.Name == pgpu.GroupName && group.ClusterId == cluster.Id))
                 .Join(_dbContext.Accounts, pgpu => pgpu.UserKerberos, account => account.Owner.Kerberos, (pgpu, account) => new { pgpu, account })
                 .Join(_dbContext.Groups, x => x.pgpu.GroupName, group => group.Name, (x, group) => new { x.pgpu, x.account, group })
-                // identify missing GroupAccounts (Left Join is via GroupJoin/SelectMany)
-                .GroupJoin(_dbContext.GroupsAccounts,
-                    x => new { AccountId = x.account.Id, GroupId = x.group.Id },
-                    ga => new { ga.AccountId, ga.GroupId },
-                    (x, groupAccounts) => new { x.pgpu, x.account, g = x.group, groupAccounts })
-                .SelectMany(x => x.groupAccounts.DefaultIfEmpty(), (x, groupAccount) => new { x.pgpu, x.account, x.g, groupAccount })
-                .Where(x => x.groupAccount == null)
-                .Select(x => new GroupAccount { GroupId = x.g.Id, AccountId = x.account.Id })
+                // identify missing GroupAccounts
+                .Where(x => !_dbContext.GroupsAccounts.Any(ga => ga.AccountId == x.account.Id && ga.GroupId == x.group.Id))
```

Comment:
Worthwhile/possible doing an await for this?

Reply:
Nope. It's part of the linq expression on `_dbContext.PuppetGroupsPuppetUsers` and translated to a query as a whole.